game/src/main/scala/io/aigar/game/Player.scala

```diff
@@ -4,23 +4,43 @@ import io.aigar.controller.response.Action
 import scala.math.round
 import com.github.jpbetz.subspace.Vector2
 
-class Player(val id: Int, startPosition: Vector2) {
+class Player(val id: Int, startPosition: Vector2) extends EntityContainer {
   var aiState: AIState = new NullState(this)
   private var currentCellId: Int = 0
   var cells = List(new Cell(currentCellId, this, startPosition))
+  var opponents = List[Player]()
 
   def update(deltaSeconds: Float, grid: Grid, players: List[Player]): Unit = {
-    if ( cells.isEmpty ) {
-      currentCellId += 1
-      cells = List(new Cell(currentCellId, this, grid.randomPosition))
+    opponents = players.filterNot(_ == this)
+    cells = handleCollision(cells, opponents).asInstanceOf[List[Cell]]
+
+    if (shouldRespawn) {
```

Comment:
Could this if block be even more generic? We already have a `shouldRespawn`, maybe we could have a `doRespawn(position)` and remove completely this part. And the `handleCollision` could be removed too. At this point, no more `update` in children.

Honnestly, we don't really care. I'm pretty sure it would be overengineering.